src/Microsoft.VisualStudio.Threading.Tests/JoinableTaskAndAsyncReaderWriterLockTests.cs

```diff
@@ -80,7 +80,7 @@ public void LockWithinRunAfterYieldMTA()
                     await Task.Yield();
                     await this.VerifyReadLockAsync();
                 });
-            }).GetAwaiter().GetResult();
+            }).WaitWithoutInlining(throwOriginalException: true);
```

Comment:
A quick explanation would help me to understand why we need change to WaitWithoutInline here (and other places too.)

Reply:
`Task.Run` merely schedules the task. And if you then synchronously block on that task, TPL will try to inline it right there if it hasn't started executing elsewhere yet. Since we're explicitly using `Task.Run` here (and elsewhere in tests) to get *off* the calling thread, inlining defeats the entire point.
This wasn't a problem before when the test was running on an STA thread because (evidently) TPL won't inline tasks on an STA thread. Now that we're moving most STA tests to MTA threads, inlining was happening everywhere and causing test failures and hangs. So this change prevents the inlining.

Reply:
I didn't know that TPL behavior.  Thanks for the information.